Update readme and docs for support versions (#497)
IntelPython/daal4py: README.md

- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py) when applied to scikit-learn from PyPi
- [![CircleCI](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master.svg?style=svg)](https://circleci.com/gh/IntelPython/daal4py/tree/test-sklearn-master) when applied to build from master branch

> ## 👁️ Follow us on Medium

Reviewer: ```suggestion
## 👀 Follow us on Medium
```

#### ℹ️ Additional limitations

Conda-Forge channel contains of the release -
[daal4py 2020.3](https://github.com/IntelPython/daal4py/releases/tag/2020.3).
Has support only Intel CPU.

</details>

### Anaconda Cloud - Intel channel
[![Conda Version](https://anaconda.org/intel/daal4py/badges/version.svg)](https://anaconda.org/intel/daal4py)

```bash
# Anaconda Cloud from Intel channel
conda install daal4py -c intel
```

<details><summary>[Click to expand] ⚠️ Package limitations </summary>

| OS / Python version     | **Python 36** | **Python 37**  | **Python 38**| **Python 39**|
| :-----------------------| :-----------: | :------------: | :-----------:| :-----------:|
|    **Linux**            |   ❌         |     ✔️         |     ❌      |      ❌      |
|    **Windows**          |   ❌         |     ✔️         |     ❌      |      ❌      |
|    **OsX**              |   ❌         |     ✔️         |     ❌      |      ❌      |

#### ℹ️ Additional limitations

Intel channel contains of the latest release -
[daal4py 2021.1](https://github.com/IntelPython/daal4py/releases/tag/2021.1).
Has support Intel CPU and GPU.

</details>

You can [build daal4py from sources](INSTALL.md) as well.

> # ⚡️ Getting Started

Reviewer: guess that getting started should be higher. So person landing on page can understand what daal4py is.

Author: Do you propose move before `Installation`?

Reviewer 2: Please also make it "Get Started" to be in line with our style guide

Author: done

You may reach out to project maintainers privately at onedal.maintainers@intel.com

# 🛠 Installation
daal4py is available at the [Python Package Index], on Anaconda Cloud in [Conda-Forge channel](https://anaconda.org/conda-forge/daal4py) and in [Intel channel](https://anaconda.org/intel/daal4py) .
```bash
# PyPi
pip install daal4py
```
```bash
# Anaconda Cloud from Conda-Forge channel
conda install daal4py -c conda-forge
```
```bash

> # Anaconda Cloud from Intel channel

Reviewer 3: I think we need to specify where conda-forge and intel channels are recommended.

Author: Done

|    **OsX**              |    [CPU]       |  [CPU]         |    [CPU]     |        ❌     |

#### 📦 Anaconda Cloud: Conda-Forge channel

| OS / Python version     | **Python 3.6** | **Python 3.7** | **Python 3.8**| **Python 3.9**|
| :-----------------------| :------------: | :------------: | :------------:| :------------:|
|    **Linux**            |   [CPU]        |   [CPU]        |     [CPU]     |     [CPU]    |
|    **Windows**          |   [CPU]        |   [CPU]        |     [CPU]     |     [CPU]    |
|    **OsX**              |   ❌          |     ❌         |     ❌       |      ❌      |

#### 📦 Anaconda Cloud: Intel channel

| OS / Python version     | **Python 3.6** | **Python 3.7** | **Python 3.8**| **Python 3.9**|
| :-----------------------| :------------: | :-------------:| :------------:| :------------:|
|    **Linux**            |   ❌         |     [CPU, GPU]   |     ❌      |      ❌      |
|    **Windows**          |   ❌         |     [CPU, GPU]   |     ❌      |      ❌      |
|    **OsX**              |   ❌         |     [CPU]        |     ❌      |      ❌      |

</details>

You can [build daal4py from sources](INSTALL.md) as well.

# ⚡️ Get Started

> Accelerate scikit-learn with the core functionality of dall4py without changing the code.

Reviewer 2: ```suggestion
Accelerate scikit-learn with the core functionality of daal4py without changing the code.
```

    clustering = DBSCAN(eps=3, min_samples=2).fit(X)
```

# 🛠 Installation
daal4py is available at the [Python Package Index](https://pypi.org/project/daal4py/),
on Anaconda Cloud in [Conda-Forge channel](https://anaconda.org/conda-forge/daal4py)
and in [Intel channel](https://anaconda.org/intel/daal4py).

### 📦 PyPi channel
[![PyPI Version](https://img.shields.io/pypi/v/daal4py)](https://pypi.org/project/daal4py/)
[![Downloads](https://pepy.tech/badge/daal4py/month)](https://pypi.org/project/daal4py/)

```bash
pip install daal4py
```
<details><summary>[Click to expand] ⚠️ Package limitations </summary>

> | OS / Python version     | **Python 36** | **Python 37**  | **Python 38**| **Python 39**|

Reviewer 2: shouldn't it be Python 3.6, Python 3.7, etc.? same Q for other tables as well

| :-----------------------| :-----------: | :------------: | :-----------:| :-----------:|
|    **Linux**            |   ✔️         |     ✔️         |     ✔️      |      ✔️      |
|    **Windows**          |   ✔️         |     ✔️         |     ✔️      |      ✔️      |
|    **OsX**              |   ✔️         |     ✔️         |     ✔️      |      ❌      |

#### ℹ️ Additional limitations

> PyPi channel contains of the latest release -

Reviewer 2: ```suggestion
PyPi channel contains the latest release:
```
or
```suggestion
PyPi channel contains the latest release,
```

[daal4py 2021.1](https://github.com/IntelPython/daal4py/releases/tag/2021.1).

> Has support Intel CPU and GPU.

Reviewer 2: not clear what you are saying, where is the limitation?

```suggestion
It supports Intel CPU and GPU.
```

</details>

> ### 📦 Anaconda Cloud - Conda-Forge channel

Reviewer 2: ```suggestion
### 📦 Anaconda Cloud: Conda-Forge channel
```

[![Conda Version](https://img.shields.io/conda/vn/conda-forge/daal4py)](https://anaconda.org/conda-forge/daal4py)
[![Downloads](https://img.shields.io/conda/dn/conda-forge/daal4py.svg)](https://anaconda.org/conda-forge/daal4py)

```bash
conda install daal4py -c conda-forge
```
<details><summary>[Click to expand] ⚠️ Package limitations </summary>

| OS / Python version     | **Python 36** | **Python 37**  | **Python 38**| **Python 39**|
| :-----------------------| :-----------: | :------------: | :-----------:| :-----------:|
|    **Linux**            |   ✔️         |     ✔️         |     ✔️      |      ✔️      |
|    **Windows**          |   ✔️         |     ✔️         |     ✔️      |      ✔️      |
|    **OsX**              |   ❌         |     ❌         |     ❌      |      ❌      |

#### ℹ️ Additional limitations

> Conda-Forge channel contains of the release -

Reviewer 2: ```suggestion
Conda-Forge channel contains the following release:
```

[daal4py 2020.3](https://github.com/IntelPython/daal4py/releases/tag/2020.3).

> Has support only Intel CPU.

Reviewer 2: ```suggestion
It supports only Intel CPU.
```

</details>

> ### 📦 Anaconda Cloud - Intel channel

Reviewer 2: ```suggestion
### 📦 Anaconda Cloud: Intel channel
```

[![Conda Version](https://anaconda.org/intel/daal4py/badges/version.svg)](https://anaconda.org/intel/daal4py)
[![Downloads](https://img.shields.io/conda/dn/intel/daal4py.svg)](https://anaconda.org/intel/daal4py)

```bash
conda install daal4py -c intel
```

<details><summary>[Click to expand] ⚠️ Package limitations </summary>

| OS / Python version     | **Python 36** | **Python 37**  | **Python 38**| **Python 39**|
| :-----------------------| :-----------: | :------------: | :-----------:| :-----------:|
|    **Linux**            |   ❌         |     ✔️         |     ❌      |      ❌      |
|    **Windows**          |   ❌         |     ✔️         |     ❌      |      ❌      |
|    **OsX**              |   ❌         |     ✔️         |     ❌      |      ❌      |

#### ℹ️ Additional limitations

> Intel channel contains of the latest release -

Reviewer 2: ```suggestion
Intel channel contains the latest release:
```
or
```suggestion
Intel channel contains the latest release,
```

[daal4py 2021.1](https://github.com/IntelPython/daal4py/releases/tag/2021.1).

> Has support Intel CPU and GPU.

Reviewer 2: ```suggestion
It supports Intel CPU and GPU.
```

    clustering = DBSCAN(eps=3, min_samples=2).fit(X)
```

# 🛠 Installation
daal4py is available at the [Python Package Index](https://pypi.org/project/daal4py/),
on Anaconda Cloud in [Conda-Forge channel](https://anaconda.org/conda-forge/daal4py)
and in [Intel channel](https://anaconda.org/intel/daal4py).

```bash
# PyPi
pip install daal4py
```

```bash
# Anaconda Cloud from Conda-Forge channel (recommended for conda users by default)
conda install daal4py -c conda-forge
```

```bash
# Anaconda Cloud from Intel channel (recommended for conda users who use other components from Intel Distribution for Python)
conda install daal4py -c intel
```

> <details><summary>[Click to expand] ℹ️ Supported configurations </summary>

Reviewer 2: I'd suggest adding a header before this line: `## Supported configurations` it'll make them more visible

> d4p.daalfini()

Reviewer 3: I think we can remove this portion about distributed algos.
Because it's not Scikit-learn patching.

Author: Done

daal4py patching affects performance of specific Scikit-learn functionality listed below. In cases when unsupported parameters are used, daal4py fallbacks into stock Scikit-learn. These limitations described below. If the patching does not cover your scenarios, [submit an issue on GitHub](https://github.com/IntelPython/daal4py/issues).

> ⚠️ We support optimizations for the last 4 versions of Scikit-learn. The latest release of daal4py-2021.1 suppors Scikit-learn 0.21, 0.22, 0.23 and 0.24.

Reviewer 3: suppors -> supports
Probably 0.21 -> 0.21.X

Author: Done

> <details><summary>[Click to expand] 🔥 Scenarios that are already available in 2020.3 release </summary>

Reviewer 3: Probably, need to rename this to better describe the content.

Author: Done

daal4py patching affects performance of specific Scikit-learn functionality listed below. In cases when unsupported parameters are used, daal4py fallbacks into stock Scikit-learn. These limitations described below. If the patching does not cover your scenarios, [submit an issue on GitHub](https://github.com/IntelPython/daal4py/issues).

> ⚠️ We support optimizations for the last 4 versions of Scikit-learn. The latest release of daal4py-2021.1 suppors Scikit-learn 0.21.X,

Reviewer: Do we really want to list all versions instead of putting statement that last 4 supported? 
It will be required to update this list with every scikit release.

Author: I'm not sure. It seems to me that we need to clearly state what we support, so that there is no question. Can we automate this?

Reviewer: Let's keep for now

Reviewer 2: ```suggestion
⚠️ We support optimizations for the last four versions of scikit-learn. The latest release of daal4py-2021.1 supports scikit-learn 0.21.X,
```

0.22.X, 0.23.X and 0.24.X.

> <details><summary>[Click to expand] 🔥 Applying the daal4py patch will impact the following existing scikit-learn algorithms </summary>

Reviewer 2: ```suggestion
<details><summary>[Click to expand] 🔥 Applying the daal4py patch will impact the following existing scikit-learn algorithms: </summary>
```

I'd move "Click to expand" after that sentence, something like this:

---
🔥 Applying the daal4py patch will impact the following existing scikit-learn algorithms:

<details><summary>[Click to expand]</summary>

|Task|Functionality|Parameters support|Data support|
|:---|:------------|:-----------------|:-----------|
|Classification|**SVC**|All parameters except `kernel` = 'poly' and 'sigmoid'. | No limitations.|

</details>

---
also not sure about fire emoji for this :)

Author: Why?)

Reviewer 2: emoji -- this is up to you, I just don't see how fire connects with the context/content

the other suggestion is just about making the hidden section a bit more visible to the readers

> 

Reviewer 3: I think we need to add one row more - when nothing is printed
